api/desecapi/models/tokens.py

```diff
@@ -167,42 +130,68 @@ class TokenDomainPolicy(ExportModelOperationsMixin("TokenDomainPolicy"), models.
+                nulls_distinct=False,
+            ),
+        ]
+        triggers = [
+            # Ensure that token_user is consistent with token (to fulfill compound FK constraint, see migration)
+            pgtrigger.Trigger(
+                name="token_user",
+                operation=pgtrigger.Update | pgtrigger.Insert,
+                when=pgtrigger.Before,
+                func="NEW.token_user_id = (SELECT user_id FROM desecapi_token WHERE id = NEW.token_id); RETURN NEW;",
+            ),
+            # Ensure that if there is *any* domain policy for a given token, there is always one with domain=None.
+            pgtrigger.Trigger(
+                name="default_policy_primacy",
+                operation=pgtrigger.Insert | pgtrigger.Update | pgtrigger.Delete,
+                when=pgtrigger.After,
+                timing=pgtrigger.Deferred,
+                func=pgtrigger.Func(
+                    """
+                    IF
+                        EXISTS(SELECT * FROM {meta.db_table} WHERE token_id = COALESCE(NEW.token_id, OLD.token_id)) AND NOT EXISTS(
+                            SELECT * FROM {meta.db_table} WHERE token_id = COALESCE(NEW.token_id, OLD.token_id) AND domain_id IS NULL AND subname IS NULL AND type IS NULL
+                        )
+                    THEN
+                        RAISE EXCEPTION 'Token policies without a default policy are not allowed.';
+                    END IF;
+                    RETURN NULL;
+                """
+                ),
             ),
         ]
 
+    @property
+    def has_default_policy_settings(self):
+        return self.domain is None and self.subname is None and self.type is None
+
     def clean(self):
-        default_policy = self.token.get_policy(domain=None)
-        if not self._state.adding:  # update
-            # Can't change policy's default status ("domain NULLness") to maintain policy precedence
-            if (self.domain is None) != (self.pk == default_policy.pk):
+        default_policy = self.token.get_policy()
+        if self._state.adding:  # create
+            # Can't violate policy precedence (default policy has to be first)
+            if (default_policy is None) and not self.has_default_policy_settings:
                 raise ValidationError(
-                    {
-                        "domain": "Policy precedence: Cannot disable default policy when others exist."
-                    }
+                    "Policy precedence: The first policy must be the default policy."
                 )
-        else:  # create
-            # Can't violate policy precedence (default policy has to be first)
-            if (self.domain is not None) and (default_policy is None):
+        else:  # update
+            # Can't change policy's default status ("domain NULLness") to maintain policy precedence
+            if self.has_default_policy_settings != (self.pk == default_policy.pk):
```

Comment:
```
is_default_policy_when_loaded = self.pk == default_policy.pk
is_default_policy_after_saving = self.has_default_policy_settings
# Can't make non-default policy default, can't make default policy non-default
if is_default_policy_when_loaded != is_default_policy_after_saving:
```

Reply:
Also update error message to catch both cases

Reply:
done